core/types/transaction_signing.go

```diff
@@ -277,6 +277,22 @@ func (sg Signer) SignatureValues(tx Transaction, sig []byte) (R, S, V *uint256.I
 	return R, S, V, nil
 }
 
+// EIP155Signer implements Signer using the EIP-155 rules. This accepts transactions which
+// are replay-protected as well as unprotected homestead transactions.
+type EIP155Signer struct {
```

Comment:
You do not need to re-introduce EIP155Signer, because it has been factored into the more generic signer type. If you want specifically EIP155 signer, you can instantiate signer with:
```
		signer.protected = true
		signer.chainID.Set(&chainId)
		signer.chainIDMul.Mul(&chainId, u256.Num2)
```

then you don't need to re-introduce EIP155Signer type